server/src/main/kotlin/fi/oph/ludos/certificate/CertificateRepository.kt

```diff
@@ -84,13 +106,19 @@ class CertificateRepository(
             sql,
             { rs: ResultSet, _: Int ->
                 FileUpload(
-                    file.fileName, file.fileKey, getZonedDateTimeFromResultSet(rs, "attachment_upload_date")
+                    fileToCreate.fileName,
+                    fileToCreate.fileKey,
+                    getZonedDateTimeFromResultSet(rs, "attachment_upload_date")
                 )
             },
-            file.fileKey,
-            file.fileName,
+            fileToCreate.fileKey,
+            fileToCreate.fileName,
         )
 
+        if (oldFileKey != null) {
```

Comment:
Pitikös tässä tsekata myös, ettei oldFileKey ole käytössä missään todistuksessa?

Jotta se tsekki ois tehokas, ja muutenkin yleisenä eheysvarmistuksena vois lisätä unique-constraintin certificate-tauluihin attachment_file_key-kolumniin. Tämä siis loisi indeksin, jonka ansiosta `SELECT foo FROM suko_certificate WHERE attachment_file_key = %`-queryt ois nopeita JA pakottaisi kantatasolla voimaan siivous-logiikan tekemän oletuksen, että yksi certificate_attachment liittyy korkeintaan yhteen certificateen.

Harmillisesti tää constraint pitäis lisätä kaikkiin kolmeen certificate-alitauluun, ku taulut eivät tosiaan peri constraintteja.

Reply:
Niinjoo, ja s3:n lisäksi pitää poistaa attachment-rivi myös kannasta.

Kannattaiskohan olla oma funkkari deleteAttachment, joka poistaa sekä s3-objektin että kantarivin?